R/addOrderSequence.R

```diff
@@ -0,0 +1,52 @@
+#' Rank the sample information in a `SummarizedExperiment` Object
+#'
+#' The information stored in `colData` in
+#' \linkS4class{SummarizedExperiment} object is ranked by `addOrderSequence`
+#' function and newly ranked field is added back to
+#' \linkS4class{SummarizedExperiment} object.
+#'
+#' @param x \linkS4class{SummarizedExperiment} object containing time series
+#' @param field A character value indicating the name of the matrix column that
+#' is aimed to be ranked.
+#' @param rank_field_name A character value to name the newly ranked `field`
+#' @param ... Allow new parameters to be defined for this function.
+#'
+#' @return \linkS4class{SummarizedExperiment} object with ranked field
+#' added
+#'
+#' @docType methods
+#' @aliases addOrderSequence-SummarizedExperiment
+#' @aliases addOrderSequence,SummarizedExperiment-method
+#'
+#' @importFrom SummarizedExperiment colData
+#' @importFrom SummarizedExperiment colData<-
+#' @importFrom methods setGeneric
+#' @importFrom TreeSummarizedExperiment TreeSummarizedExperiment
+#'
+#' @examples
+#' data(GlobalPatterns, package="mia")
+#' se <- GlobalPatterns
+#'
+#' se1 <- addOrderSequence(se,field = "SampleType",
+#'             rank_field_name = "SampleType_rank",
+#'             na.last = TRUE, ties.method = "first")
+#'
+#' #example 2: new matrix and its rank added
+#'
+#' se$newfield <- matrix(1:26, nrow = 26, ncol = 1)
+#' se2 <- addOrderSequence(se, field = "newfield" ,
+#'                 rank_field_name = "newfield_rank" , na.last = TRUE ,
+#'                 ties.method = "first")
```

Comment:
Could we also have example with a numeric or date field for `addOrderSequence`? Either from GlobalPatterns or from hitchip1006.